scripts/utils.ts

```diff
@@ -46,13 +63,59 @@ export async function compileContracts(): Promise<CompilerOutput> {
 	return compilerOutput
 }
 
+function getScriptErrorMessage(errorCode: ScriptErrorCode, errorParameters?: string[]): string {
+	switch (errorCode) {
+		case ScriptErrorCode.RPC_URL_NOT_FOUND:
+			return `**⛔️ Error:**<br>RPC URL not found in the issue body.`
+		case ScriptErrorCode.FACTORY_ALREADY_DEPLOYED:
+			return `**⛔️ Error:**<br>The factory is already deployed.`
+		case ScriptErrorCode.CHAIN_NOT_LISTED:
+			return `**⛔️ Error:**<br>Chain ${errorParameters?.[0]} is not listed in the chainlist. For more information on how to add a chain, please refer to the [chainlist repository](https://github.com/ethereum-lists/chains).<br>`
+		case ScriptErrorCode.FACTORY_DIFFERENT_BYTECODE:
+			return `**⛔️ Error:**<br>Factory is deployed with different bytecode.`
+		case ScriptErrorCode.FACTORY_PRE_DEPLOYED:
+			return `**⛔️ Error:**<br>Factory is pre-deployed on the chain.`
+		case ScriptErrorCode.FACTORY_NOT_ADDED_TO_REPO:
+			return `**⛔️ Error:**<br>Factory has been deployed but not added to the repository.`
+		case ScriptErrorCode.FACTORY_DEPLOYER_ACCOUNT_NONCE_BURNED:
+			return `**⛔️ Error:**<br>Factory deployer account nonce burned.`
+		case ScriptErrorCode.GAS_PRICE_NOT_RETRIEVED:
+			return `**⛔️ Error:**<br>Gas price couldn't be retrieved. Please make sure that the RPC URL is valid and reachable.`
+		case ScriptErrorCode.GAS_LIMIT_NOT_ESTIMATED:
+			return `**⛔️ Error:**<br>Gas limit couldn't be estimated. Please make sure that the RPC URL is valid and reachable.`
+		case ScriptErrorCode.GAS_LIMIT_ESTIMATION_FAILED:
+			return `**⛔️ Error:**<br>Gas limit estimation failed. Please make sure that the RPC URL is valid and reachable.`
+		case ScriptErrorCode.DEPLOYMENT_SIMULATION_FAILED:
+			return `**⛔️ Error:**<br>Deployment simulation failed. Please make sure that the RPC URL is valid and reachable.`
+		case ScriptErrorCode.FACTORY_DEPLOYMENT_SIMULATION_DIFFERENT_BYTECODE:
+			return `**⛔️ Error:**<br>Factory deployment simulation returned different bytecode.`
+		case ScriptErrorCode.PREFUND_NEEDED:
+			return `**💸 Pre-fund needed:**<br/>We need a pre-fund to deploy the factory. Please send ${errorParameters?.[0]} wei to ${SIGNER} and check the checkbox in the issue.`
+		default:
+			return `**⛔️ Error:**<br>Unknown error`
+	}
+}
+
+export class ScriptError extends Error {
+	constructor(public exitCode: number = 1, errorParameters?: string[]) {
```

Comment:
nit: This should be a common way for scripts to report different error codes. I really believe it should be:

```suggestion
	constructor(message: string, public exitCode: number = 1) {
```

Whatever additional logic (like generating the `new-chain` specific error messages) should be part of `new-chain` script, and not common utilities.

Reply:
Updated 👍🏾